Formula/p/pyenv.rb

```diff
@@ -1,9 +1,11 @@
 class Pyenv < Formula
   desc "Python version management"
   homepage "https://github.com/pyenv/pyenv"
-  url "https://github.com/pyenv/pyenv/archive/refs/tags/v2.4.5.tar.gz"
-  sha256 "24f3671a92492f49fa6f61dea861323ad853877c5cb2686c95bfa76339aa1301"
+  url "https://github.com/pyenv/pyenv/archive/2c2c7616e2c3a8ab7623bea3c353ff520750b812.tar.gz"
+  version "2.4.5"
```

Comment:
we'd like to use tag rather than specific commit

Reply:
That's impossible. I recreated the release, https://github.com/pyenv/pyenv/archive/refs/tags/v2.4.5.tar.gz has new content but https://codeload.github.com/pyenv/pyenv/tar.gz/refs/tags/v2.4.5 that `brew install` downloads from instead has old content, and it's impossible to force it to regenerate.

Reply:
If I understand correctly you tagged  a `v2.4.5` release and later moved that tag to a different commit, which is very bad practice, begging for such problems.

I guess best course of action is to create a *new* `v2.4.6` release / tag.
You could move the tag `v2.4.5` back to where it was (and hope nobody uses it and it wasn't cached anywhere with the other commit), or just delete it.

Reply:
That said, I can create a tag, say, `v2.4.5_1`, with the same internal version number, and substitute it here. Would that do?

Reply:
> In any case, I cannot reproduce the caching problem you describe:

I checked with `curl https://github.com/pyenv/pyenv/archive/refs/tags/v2.4.5.tar.gz |sha256sum` and got a different checksum of `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`. Apparently, that's the checksum for empty content.

Reply:
yes, because it's a redirect so you need `-L`

Reply:
> I don't want the project's history to be a hall of shame of past mistakes. The mistake didn't affect functionality, only the reported version -- so a new version number is unwarranted.
> 
> A formula revision mechanism seems to exactly be designed for situations where a version number change is not warranted -- so that's what I did. Badly designed caching tools are not a reason to tolerate paying for their flaws from your own pocket (in the form of mutilated project history).

Please read the git manual's section on [re-tagging](https://git-scm.com/docs/git-tag#_on_re_tagging), and why it's the "insane thing" to do. I understand the desire to have a clean repo history, but once you've released a tag it shouldn't be touched again. The hash for the original tag was consumed and recorded by multiple downstream sources. This introduces a security issue where consumers are unsure if something malicious has happened upstream. I urge you to reconsider re-tagging for the sake of your users.

Reply:
@native-api it is really about shame, the human errors happen all the time, it would be just roll forward with a new release IMO